### ISSUE TITLE
include secrets in wrangler publish

### ISSUE THREAD

**Issue by ashleymichal** [cloudflare/wrangler#911]
If we decide that the api should continue requiring all bindings to be declared on every upload, Wrangler will need to know the names of secret bindings in order to include them in the upload.

### Syntax
Users will need to declare a top level list of secret bindings by name:

```toml
name = "worker"
type = "javascript"
workers_dev = true
secrets = ["THIRD_PARTY_API_KEY", "PRIVATE_KEY"]
[env.prod]
#...
```

### Implementation
For each secret name listed, Wrangler will need to add an entry to the bindings list in the metadata form it sends on upload; the entry can simply contain the `name` of the binding, with `type: inherit`.

### Environments
Because a wrangler.toml should not contain the secret values themselves, and because the names of the secrets will remain consistent between environments (they will all be called by the same name in the script), it makes sense that the secret configuration should be a top-level key. However, this does not imply that the value of the key will be the same between environments; users are encouraged to set secrets per environment using the secret create command ( #907 ) or via the Cloudflare Dashboard.

TL;DR - wrangler should complain if the user tries to override the secrets config at the env level.

### On error
Since wrangler is simply telling the API to use whatever secret currently exists for a given script by a given name, it is possible that wrangler publish will return with an error complaining that one or more bindings does not exist. We can include those in the output, and suggest the correct command to create the given secret(s) for the environment in question.

### Depends on
this issue assumes that no changes will be made to the logic around bindings in the config api, however some error enhancements will make this easier to deal with, specifically:

* the API should return a specific error listing any missing "inherit" bindings.

**Comment by lolcoolkat** [comment]
Super excited about this!

**Comment by ashleymichal** [comment]
closing as won't do. the api will be updated to preserve secrets uploaded to a script until they are explicitly deleted. wrangler will instead support commands to create (#907)  and delete (#912)  secrets for a script.
